bitnami/postgresql/values.yaml

```
@@ -19,14 +19,23 @@ global:
     ## @param global.postgresql.auth.username Name for a custom user to create (overrides `auth.username`)
     ## @param global.postgresql.auth.password Password for the custom user to create (overrides `auth.password`)
     ## @param global.postgresql.auth.database Name for a custom database to create (overrides `auth.database`)
-    ## @param global.postgresql.auth.existingSecret Name of existing secret to use for PostgreSQL credentials (overrides `auth.existingSecret`)
+    ## @param global.postgresql.auth.existingSecret.enabled Enable the use an existing secret for PostgreSQL credentials (overrides `auth.existingSecret`).
+    ## @param global.postgresql.auth.existingSecret.secretName Name of existing secret to use for PostgreSQL credentials.
+    ## @param global.postgresql.auth.existingSecret.adminPasswordKey Name of key in existing secret to use for PostgreSQL credentials.
+    ## @param global.postgresql.auth.existingSecret.userPasswordKey Name of key in existing secret to use for PostgreSQL credentials.
+    ## @param global.postgresql.auth.existingSecret.replicationPasswordKey Name of key in existing secret to use for PostgreSQL credentials.
     ##
     auth:
       postgresPassword: ""
       username: ""
       password: ""
       database: ""
-      existingSecret: ""
+      existingSecret:
+        enabled: false
+        secretName: name-of-secret
+        adminPasswordKey: admin-password
+        userPasswordKey: user-password
+        replicationPasswordKey: replication-password
```

In this case we can add this new feature without breaking previous versions. For example:

```suggestion
      existingSecret: ""
      secretKeys:
        adminPasswordKey: admin-password
        userPasswordKey: user-password
        replicationPasswordKey: replication-password
```

Keep using `existingSecret` as name for the secret and adding a new parameter for the different keys. With this approach user don't need to change their values files to keep working and we don't need to upgrade all charts depending on postgresql

---

That's a good idea, thanks for that! I'll try to implement that.